pyiso/isone.py

```diff
@@ -188,3 +204,66 @@ def parse_json_load_data(self, data):
                 return data['FiveMinSystemLoads']['FiveMinSystemLoad']
         except (KeyError, TypeError):
             raise ValueError('Could not parse ISONE load data %s' % data)
+
+    def parse_json_lmp_data(self, data):
+        """
+        Pull approriate keys from json data set.
+        Raise ValueError if parser fails.
+        """
+        try:
+            if self.options.get('latest'):
+                return data['FiveMinLmp']
+            else:
+                return data['FiveMinLmps']['FiveMinLmp']
+        except (KeyError, TypeError):
+            raise ValueError('Could not parse ISONE lmp data %s' % data)
+
+    def get_lmp(self, zone_name, latest=True, start_at=False, end_at=False, **kwargs):
+        # set args
+        self.handle_options(data='lmp', latest=latest,
+                            start_at=start_at, end_at=end_at, **kwargs)
+
+        # get location id
+        try:
+            locationid = self.locations[zone_name.upper()]
+        except KeyError:
+            raise ValueError('No LMP data available for location %s' % zone_name)
+
+        # set up storage
+        raw_data = []
+        parsed_data = []
+
+        # collect raw data
+        for endpoint in self.request_endpoints(locationid):
+            # carry out request
+            data = self.fetch_data(endpoint, self.auth)
+
+            # pull out data
+            try:
+                raw_data += self.parse_json_lmp_data(data)
+            except ValueError as e:
+                logger.warn(e)
+                continue
+
+        # parse data
+        for raw_dp in raw_data:
+            # set up storage
+            parsed_dp = {}
+
+            # add values
+            parsed_dp['timestamp'] = self.utcify(raw_dp['BeginDate'])
```

Comment:
Either gets the job done, of course :). But for consistency with the
current format of get_load, etc, the list of dicts is better. It's my bad
for not catching that when we did CAISO get_lmp. In fact, should copy
test_load.py and make a similar test_lmp.py, that's an easy way to enforce
consistency.
On Aug 12, 2015 7:35 AM, "ndavis6" <notifications@github.com> wrote:

> In pyiso/isone.py
> <https://github.com/WattTime/pyiso/pull/31#discussion_r36866739>:
>
> > +            data = self.fetch_data(endpoint, self.auth)
> > +
> > +            # pull out data
> > +            try:
> > +                raw_data += self.parse_json_lmp_data(data)
> > +            except ValueError as e:
> > +                logger.warn(e)
> > +                continue
> > +
> > +        # parse data
> > +        for raw_dp in raw_data:
> > +            # set up storage
> > +            parsed_dp = {}
> > +
> > +            # add values
> > +            parsed_dp['timestamp'] = self.utcify(raw_dp['BeginDate'])
>
> caiso get_lmp return a dict of prices, this returns a list of dicts. Is
> one preferable?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WattTime/pyiso/pull/31/files#r36866739>.
>